internal/provider/peering.go

```diff
@@ -13,10 +13,24 @@ var peeringDefaultTimeout = time.Minute * 1
 var peeringCreateTimeout = time.Minute * 35
 var peeringDeleteTimeout = time.Minute * 35
 
-func parsePeeringResourceID(id string) (hvnID, peeringID string, err error) {
-	idParts := strings.SplitN(id, ":", 2)
-	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
-		return "", "", fmt.Errorf("unexpected format of ID (%q), expected {hvn_id}:{peering_id}", id)
+func parsePeeringResourceID(resourceID, clientProjectID string) (projectID, hvnID, peeringID string, err error) {
+	idParts := strings.SplitN(resourceID, ":", 3)
+
+	if len(idParts) == 3 { // {project_id}:{hvn_id}:{peering_id}
```

Comment:
I wonder if counting the number of `:` before splitting could help reduce some of the nested conditionals here 🤔

Reply:
In this case I feel as thought I may have the same number of conditionals since I also want to validate that none of the fields are empty.